### PR TITLE
Fixed #5863 - ZIOStream 2.0 from javaStream

### LINKED ISSUE #5863: ZIO Streams 2.0: fromJavaStream constructors
Some of the changes to the JVM-specific `fromJavaStream` constructors of `ZStream` were not yet migrated to the ZIO Streams 2.0 implementation. 

Check that all the variants are defined in `zio.stream.experimental.ZStreamPlatformSpecific`. This involves defining some new aliases and deprecate the old methods.

### PR DESCRIPTION
**Pull request by stsatlantis**
- Adds missing deprecated methods
- Adds jvm-specific implicit conversion

- removed Rawr 😄

### CODE REVIEW

File: streams-tests/jvm/src/test/scala/zio/stream/experimental/ZStreamPlatformSpecificSpec.scala [hunk 1]
```diff
@@ -103,7 +103,6 @@ object ZStreamPlatformSpecificSpec extends ZIOBaseSpec {
             stream = ZStream.asyncZIO[Any, Throwable, Int](
                        cb => {
                          global.execute { () =>
-                           println("Rawr")
```
Comment (vigoo) [inline streams-tests/jvm/src/test/scala/zio/stream/experimental/ZStreamPlatformSpecificSpec.scala:106]: 😮

Comment (stsatlantis) [reply streams-tests/jvm/src/test/scala/zio/stream/experimental/ZStreamPlatformSpecificSpec.scala:106]: If it was there for a reason I can revert this change

Comment (vigoo) [reply streams-tests/jvm/src/test/scala/zio/stream/experimental/ZStreamPlatformSpecificSpec.scala:106]: I'm sure there wasn't :)

**Review by vigoo — APPROVED** [review]
Looks good, thank you!

**Pull request merged**